### PR TITLE
feat/US34309: [UI] Text and Upgrade contract governance proposals

### PR DESCRIPTION
**Pull request by anshulkhare-lab49**
**Description**:
Added Support for Text and Upgrade Contract Proposals
-  Fetched Proposals fro text and upgrade type.
-  added turnout in proposal card.
-  updated contract json to add support for proposals

Screenshot 2023-07-05 at 2 28 33 PM

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by damiano-melcarne-lab49 — CHANGES_REQUESTED** [review]
1) Have you tested the end-to-end flow for creating, viewing, and executing `Text` and `Upgrade Contract` proposals?

2) For this and future PRs, please add UI screenshots that highlight the core set of new features/changes introduced. This helps verify if the changes on a local running copy of the UI match the intended changes while reviewing the code.

File: src/dex-ui/services/MirrorNodeService/MirrorNodeService.ts [hunk 1]
```diff
@@ -221,31 +222,36 @@ function createMirrorNodeService() {
...
+      params,
     });
 
     const allEvents = decodeLog(abiSignatures, response.data.logs, ["ProposalDetails"]);
     const proposalCreatedEvents = allEvents.get("ProposalDetails") ?? [];
     const proposals: MirrorNodeDecodedProposalEvent[] = proposalCreatedEvents.map((item: any) => {
       return { ...item, contractId, type: proposalType };
     });
-    return proposals;
+    const uniqueProposals = uniqBy((proposal: MirrorNodeDecodedProposalEvent) => proposal.proposalId, proposals);
```
[the remark this answers is not in the document]
Comment (anshulkhare-lab49) [reply src/dex-ui/services/MirrorNodeService/MirrorNodeService.ts:253]: When user votes on any proposal, a new proposal detail event is generated with same proposal id. We need to show only the latest one, so removed the old ones with UniqBy.

File: src/dex-ui/pages/dao/DAOProposalVoting.tsx [hunk 2]
```diff
@@ -11,7 +11,7 @@ interface DAOProposalVotingProps {
 export const DAOProposalVoting = (props: DAOProposalVotingProps) => {
```
[the remark this answers is not in the document]
Comment (anshulkhare-lab49) [reply src/dex-ui/pages/dao/DAOProposalVoting.tsx:11]: Done.

File: src/dex-ui/hooks/dao/useGovernanceDAOProposals.tsx [hunk 3]
```diff
@@ -57,24 +57,29 @@ export function useGovernanceDAOProposals(
...
-        .div(convertEthersBigNumberToBigNumberJS(max ?? BigNumber.from(1)))
-        .multipliedBy(convertEthersBigNumberToBigNumberJS(BigNumber.from(100)))
-        .toFixed(3)
-    );
+    const remaining = max - totalVotes;
+    const quorum = max
+      ? (convertEthersBigNumberToBigNumberJS(BigNumber.from(proposalData.votingInformation?.quorumValue ?? 0))
+          .shiftedBy(-precisionValue)
+          .toNumber() *
+          100) /
```
[the remark this answers is not in the document]
Comment (anshulkhare-lab49) [reply src/dex-ui/hooks/dao/useGovernanceDAOProposals.tsx:71]: Made the change in the flow as per suggestions.

File: src/dex-ui/services/MirrorNodeService/MirrorNodeService.ts [hunk 4]
```diff
@@ -221,31 +222,36 @@ function createMirrorNodeService() {
...
-         * To reduce the frequency at which this error occurs a temporary limit has been put in place. A
-         * max of 5 proposal details for each type of proposal will be fetched from the proposal logs.
-         *
-         * @see {@link file://./../../../../architecture/05_Event_Based_Historical_Queries.md} for more details
-         * regarding a long term solution.
-         */
-        limit: 5,
-      },
+      params,
     });
```
Comment (roshan-bisht-lab49) [inline src/dex-ui/services/MirrorNodeService/MirrorNodeService.ts:246]: We already had fetchEvents `function` in place any special case we needed this one.?

File: src/dex-ui/services/MirrorNodeService/MirrorNodeService.ts [hunk 5]
```diff
@@ -221,31 +222,36 @@ function createMirrorNodeService() {
...
     const proposalCreatedEvents = allEvents.get("ProposalDetails") ?? [];
     const proposals: MirrorNodeDecodedProposalEvent[] = proposalCreatedEvents.map((item: any) => {
       return { ...item, contractId, type: proposalType };
     });
-    return proposals;
+    const uniqueProposals = uniqBy((proposal: MirrorNodeDecodedProposalEvent) => proposal.proposalId, proposals);
+    return uniqueProposals;
   };
 
   const fetchUpgradeContractEvents = async (contractId: string, userID: string): Promise<BigNumber | undefined> => {
```
Comment (roshan-bisht-lab49) [inline src/dex-ui/services/MirrorNodeService/MirrorNodeService.ts:257]: Fetching Locked GOD tokens in separate hook which can handle the user specific amounts. we might need to refactor this part I will address this in my upcoming task which is related to this one.

File: src/dex-ui/services/contractsUAT.json [hunk 6]
```diff
@@ -204,13 +195,31 @@
...
+  {
+    "name": "nftdaofactory",
+    "id": "0.0.15021409",
+    "address": "0x0000000000000000000000000000000000e53561",
+    "timestamp": "2023-06-30T10:53:46.503Z",
+    "hash": "91b53d6e773b56141cef7ebd77d8853d",
+    "transparentProxyAddress": "0x0000000000000000000000000000000000e53565",
+    "transparentProxyId": "0.0.15021413"
   }
 ]
```
Comment (roshan-bisht-lab49) [inline src/dex-ui/services/contractsUAT.json:225]: 1. 
Screenshot 2023-07-05 at 3 03 23 PM

Quorum is not visible in UI.

2. 
Screenshot 2023-07-05 at 3 00 43 PM

I am not able to set Quorum as 0.1% of total supply minimum is 1%.

3.
Screenshot 2023-07-05 at 2 57 35 PM

Created By is missing in UI.

4. 
Screenshot 2023-07-05 at 2 57 19 PM
Under Details section we can remove Title as it coming in Top (As per Figma)

**Review by roshan-bisht-lab49 — CHANGES_REQUESTED** [review]
Just suggestions

File: src/dex-ui/services/contractsUAT.json (hunk 6 above)
Comment (anshulkhare-lab49) [reply src/dex-ui/services/contractsUAT.json:225]: 1. Added Quorum in the Horizontal chart, and used the same horizontal chart we are using in Governance.
2. Quorum is incremented in 1/100th of the percentage. added logic to convert value while creating DAO.
3. Added the created by field.
4. In details it is not title, it is proposal type. @damiano-melcarne-lab49 created it. so I used it as it is.  Let me know if we want to remove it. This is the only place where proposal type is visible in UI.

Comment (anshulkhare-lab49) [reply src/dex-ui/services/contractsUAT.json:225]: Screenshot 2023-07-06 at 4 00 22 PM
Screenshot 2023-07-06 at 4 01 57 PM
Screenshot 2023-07-06 at 4 02 52 PM

**Review by roshan-bisht-lab49 — APPROVED** [review]

File: src/dex-ui/services/contractsUAT.json (hunk 6 above)
Comment (damiano-melcarne-lab49) [reply src/dex-ui/services/contractsUAT.json:225]: @anshulkhare-lab49 
- Leave the proposal type for now. We need to holistically rethink the details section design. We can fix this in a future task.
- Convert the "created" address to a Hedera Account Id (`AccountId.fromSolidityAddress(created).toString()`). You can then use the `HashScanLink` component to convert the account id text into a clickable link.

**Review by damiano-melcarne-lab49 — CHANGES_REQUESTED** [review]
https://github.com/hashgraph/hedera-accelerator-defi-dex-ui/pull/185#discussion_r1255089748

File: src/dex-ui/services/contractsUAT.json (hunk 6 above)
Comment (anshulkhare-lab49) [reply src/dex-ui/services/contractsUAT.json:225]: Removed the transaction field as in Gov DAO we don't have transaction hash.
Removed the created field as author is already shown in the the top.
Changed Style for Type. 
Screenshot 2023-07-07 at 4 40 43 PM

**Review by damiano-melcarne-lab49 — APPROVED** [review]

**Pull request merged**